src/models/ComponentStyle.js

```diff
@@ -36,15 +34,17 @@ export default (nameGenerator: NameGenerator) => {
      * Returns the hash to be injected on render()
      * */
     generateAndInjectStyles(executionContext: Object) {
-      const flatCSS = flatten(this.rules, executionContext).join('')
-        .replace(/^\s*\/\/.*$/gm, '') // replace JS comments
-      const hash = hashStr(this.componentId + flatCSS)
+      const flatCSS = flatten(this.rules, executionContext)
+      const hash = hashStr(this.componentId + flatCSS.join(''))
+
```

Comment:
Here a little error is hidden.
First you check **internal** hash, and next add selector to a **global** css scope. 
What next? You can just duplicate style in two different style tags.